go/tools/gazelle/packages/package.go

```diff
@@ -15,7 +15,46 @@ limitations under the License.
 
 package packages
 
-import "strings"
+import (
+	"fmt"
+	"strings"
+)
+
+// PlatformConstraints is a map from config_setting labels (for example,
+// "@io_bazel_rules_go//go/platform:linux_amd64") to a sets of build tags
+// that are true on each platform (for example, "linux,amd64").
+type PlatformConstraints map[string]map[string]bool
+
+// ParsePlatformConstraints parses a string that contains all platforms and
+// tags on each platform. The string must contain an even number of fields,
+// separated by whitespace. Even fields (counting from 0) are platform labels.
```

Comment:
This won't necessarily be a temporary solution, so it would be good to get this right early on. Agree that whitespace is a bad idea. No opinion about whether tags or labels should be first.

It should be rare that users need to set this flag manually. It should only be needed if they want to support fewer platforms (for example, they don't want any mention of `linux` in their BUILD files) or if they want to add custom `config_setting` rules, implied by a custom set of tags.

Instead of +, how do you feel about semicolons?

```
linux,amd64=@io_bazel_rules_go//go/platform:linux_amd64;darwin,amd64=@io_bazel_rules_go//go/platform:darwin_amd64
```

As we add more options to Gazelle, especially complicated options like this, we might want to consider supporting per-project configuration files.